experiments/compositions/composition/internal/controller/expander_reconciler.go

```diff
@@ -126,21 +122,59 @@ func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 				Stages: map[string]*compositionv1alpha1.StageStatus{},
 			},
 		}
-		if err := ctrl.SetControllerReference(&inputcr, &plancr, r.Scheme); err != nil {
+		if err := ctrl.SetControllerReference(inputcr, &plancr, r.Scheme); err != nil {
 			logger.Error(err, "Unable to set controller reference for Plan Object")
-			return ctrl.Result{}, err
+			return planNN, &plancr, err
 		}
 		if err := r.Client.Create(ctx, &plancr); err != nil {
 			logger.Error(err, "Unable to create Plan Object")
-			return ctrl.Result{}, err
+			return planNN, &plancr, err
 		}
 
 		// Read after create to get the UID
 		if err := r.Client.Get(ctx, planNN, &plancr); err != nil {
 			logger.Error(err, "Unable to fetch Plan Object")
+			return planNN, &plancr, err
+		}
+	}
+	return planNN, &plancr, nil
+}
+
+func (r *ExpanderReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger = logger.WithName(r.Composition.Name).WithName(r.InputGVK.Group)
+
+	logger.Info("Got Input API for expansion", "request", req)
+
+	inputcr := unstructured.Unstructured{}
+	inputcr.SetGroupVersionKind(r.InputGVK)
+	if err := r.Get(ctx, req.NamespacedName, &inputcr); err != nil {
+		logger.Error(err, "unable to fetch Input API Object")
```

Comment:
We should indicate which API object we could not fetch.

Reply:
done.